### PR TITLE
Airflow stack update to v2.9.0

### LINKED ISSUE #16921: Upgrade Airflow Stack to >= 2.9.0
We currently run Airflow 2.6.1, and it was release on May 16, 2023. The current version is 2.9.0, and I would like to upgrade our stack to this version. Small incremental upgrades are more likely to be easy, and I would like to get this important stack updated before it becomes a headache.

### PR DESCRIPTION
**Pull request by frankhereford**
## TLDR

This was super easy. I browsed the major feature changes on the READMEs of the releases since our initial install, and nothing jumped out at me. I was able to make the update work with only one tweak -- installing `gosu`. 

## Associated issues

This PR intends to close https://github.com/cityofaustin/atd-data-tech/issues/16921.

## Associated repo

[atd-airflow](https://github.com/cityofaustin/atd-airflow)

### Cluster Activity and Health page

image

## Testing

* Checkout this branch
* Edit [this line](https://github.com/cityofaustin/atd-airflow/blob/airflow-update-2.9.0/docker-compose.yaml#L9) to something like `  image: atddocker/atd-airflow:testing`, noting the tag you choose. It can be anything, this won't get pushed anywhere.
* Build the new airflow image with the following command, making sure to use the same tag you chose.
```bash
docker build -t atddocker/atd-airflow:testing .
```
* Fire up airflow like normal with `docker compose up`, and this terminal becomes the logs to watch while you test. 
* Visit your [local airflow instance](http://localhost:8080/home?status=all)
* Verify the version is `2.9.0` in the footer
* And test anything and everything you want to - everything is in bounds.
  * One DAG I felt fine about firing up off over and over is the `maximo_geo_emergency_mgmt_email_parser_development` one.
* ⚠️ Don't forget to undo the change you made to your docker compose file.
---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Add note to 1PW secrets moved to API vault and check for duplicates

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -2,7 +2,6 @@
 # https://airflow.apache.org/docs/apache-airflow/2.6.1/docker-compose.yaml
 # and documented here:
 # https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html#running-airflow-in-docker
-version: "3.8"
```
Comment (frankhereford) [inline docker-compose.yaml:5]: Tag-along change. The docker project dropped support for this designation.

**Review by mddilley — APPROVED** [review]
Awesome! I tested this locally with your instructions, and it worked perfect! Thanks for upgrading and pointing out the new dashboard too! 🙏 🚢

**Review by chiaberry — APPROVED** [review]
thanks for keeping us up to date!
